crates/ide-db/src/path_transform.rs

```diff
@@ -248,6 +248,47 @@ fn preorder_rev(item: &SyntaxNode) -> impl Iterator<Item = SyntaxNode> {
     x.into_iter().rev()
 }
 
+fn is_path_in_expression_context(path: &ast::Path) -> bool {
+    let mut current = path.syntax().parent();
+    while let Some(node) = current {
+        // Expression contexts where we need bare type names
+        if let Some(call_expr) = ast::CallExpr::cast(node.clone()) {
+            if let Some(expr) = call_expr.expr() {
+                if expr.syntax().text_range().contains_range(path.syntax().text_range()) {
+                    return true;
+                }
+            }
+        }
+        if ast::PathExpr::cast(node.clone()).is_some() {
+            return true;
+        }
+        if let Some(record_expr) = ast::RecordExpr::cast(node.clone()) {
+            if let Some(record_path) = record_expr.path() {
+                if record_path.syntax().text_range().contains_range(path.syntax().text_range()) {
+                    return true;
+                }
+            }
+        }
+        // Stop at type/pattern boundaries
+        if ast::Type::cast(node.clone()).is_some()
+            || ast::Pat::cast(node.clone()).is_some()
+            || ast::RetType::cast(node.clone()).is_some()
+        {
+            return false;
+        }
+        current = node.parent();
+    }
+    false
+}
```

Comment:
This can be simplified (and also made more robust by that)
```suggestion
fn is_path_in_expression_context(path: &ast::Path) -> bool {
	let path = path.top_path();
    let Some(parent) = path.syntax().parent() else { return false; };
    parent.kind() == SyntaxKind::PATH_EXPR
}
```
I don't think we need to consider record expression constructors, as their path is actually in type namespace if I am not mistaken.

Reply:
Okay, my original concern was about struct constructor calls like `Foo { field: value }` where we might want Foo instead of `Self::Foo`, but thinking about it more, record expressions are indeed type namespace references and even if we wanted bare names there, `PathTransform` should handle the namespace correctly, so this complexity isn't justified indeed, thanks.

Reply:
We can always add back heuristics later that can scan for cases where we can simplify the path, but for now that's not really necessary I think